daphne_worker/src/dap.rs

```diff
@@ -297,12 +356,27 @@ impl<D> DapLeader<BearerToken> for DaphneWorkerConfig<D> {
             .get_task_config_for(&collect_req.task_id)
             .ok_or_else(|| DapError::fatal(INT_ERR_UNRECOGNIZED_TASK))?;
 
+        if collect_req.batch_interval.duration / task_config.min_batch_duration
+            > self.global_config.max_batch_duration
+            || now().abs_diff(collect_req.batch_interval.start)
+                > self.global_config.min_batch_interval_start
+            || now().abs_diff(collect_req.batch_interval.end())
+                > self.global_config.max_batch_interval_end
+        {
+            return Err(DapError::Abort(DapAbort::InvalidBatchInterval));
```

Comment:
Just as for `MockAggregator`, this should return something like `DapAbort::BadRequest("batch interval ")`. You might also consider leaving a note here that we might consider addressing in the spec.

Reply:
Changed!